app/src/workers/soloWorker.js

```diff
@@ -0,0 +1,43 @@
+import MissedPingsMq from '../db/MissedPingsMq.js';
+import { dbGetMonitorById, dbAddRun, dbUpdateMonitorFailing } from '../db/queries.js';
+import { nextScheduledRun } from '../utils/cronParser.js';
+
+const handleMissingMonitor = (monitor) => {
+  if (!monitor) {
+    const error = new Error('soloWorker: No monitor associated with that id.');
+    throw error;
+  }
+};
+
+const calculateDelay = (monitor) => {
+  const runTime = nextScheduledRun(monitor.schedule)._date.ts +
+    ((monitor.grace_period + monitor.tolerable_runtime) * 1000); // milliseconds from epoch
+
+  return (runTime - Date.now()) / 1000; // delay in seconds
+};
+
+const startWorker = async (job) => {
+  console.log('soloWorker triggerd: ', job);
+  try {
+    const monitor = await dbGetMonitorById(job.data.monitorId);
+    handleMissingMonitor(monitor);
+
+    if (!monitor.failing) {
```

Comment:
Should the monitor state be set to failing from the workers? Where else would that happen from?

Reply:
Yeah I think it needs to be set from here because the job triggering means a missed ping which could take us from a passing to a failing state - I'm not sure where else we could put it. 

State changes will also occur in the `pings` controller - specifically on `solo`, `ending` and `failing` pings depending on the previous state of the monitor. I am implementing this now.